### PR TITLE
Single deletion

### LINKED ISSUE #590: Implement Single Deletion (text)
Implementing a method that deletes single words from a sentence and measure the effect on the predictive model. possibly compute correlation with the given relevances.

### LINKED ISSUE #599: Extend Incremental Deletion metric with random baseline.

### PR DESCRIPTION
**Pull request by WillemSpek**
Implemenation of single Deletion for text methods. Tutorial is included in the `metrics_text.ipynb`. Example visualization is given below: 

LIMEtext

### CODE REVIEW

**Review by elboyran — COMMENTED** [review]
Looks OK, did you address Chri's comments? Also, I see a type error in one notebook, does it actually run successfully?

File: relevance_maps_properties/metrics/metrics_text.ipynb [hunk 1]
```diff
@@ -0,0 +1,300 @@
+{
```
Comment (cwmeijer) [inline relevance_maps_properties/metrics/metrics_text.ipynb:1]: <p>Why do you conclude this for RISE? I only see LIME being used in this notebook.</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/dianna-ai/dianna-exploration/pull/163/discussion/#comment-1209992579'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

File: relevance_maps_properties/metrics/metrics_tutorial.ipynb [hunk 2]
```diff
@@ -0,0 +1,369 @@
+{
```
Comment (cwmeijer) [inline relevance_maps_properties/metrics/metrics_tutorial.ipynb:1]: <p>This is an awesome figure!</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/dianna-ai/dianna-exploration/pull/163/discussion/#comment-1209992588'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (cwmeijer) [inline relevance_maps_properties/metrics/metrics_tutorial.ipynb:1]: <p>I don't understand this Figure. What is compared with what? At 40% removal, we see a dramatic drop. This means, that after removing 40% of the most important parts according to our XAI method, the models performance suddenly drops. Is this correct? We already saw the same thing in the Figure above. What does this Figure add then? If it's just another (maybe more detailed) visualization of the same thing, then say so in the comments. It's now a little surprising. Also, in that case, I would have expected a second data series in the same plot showing the 'random order' deletion strategy.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/dianna-ai/dianna-exploration/pull/163/discussion/#comment-1209992594'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

**Review by cwmeijer — CHANGES_REQUESTED** [review]
See comments in the bee and text notebooks

File: relevance_maps_properties/metrics/metrics_tutorial.ipynb (hunk 2 above)
Comment (WillemSpek) [reply relevance_maps_properties/metrics/metrics_tutorial.ipynb:1]: Idea of the images is to give the user a visual idea of what's going on. What is actually interesting is the area under the deletion curve or between the deletion and the random curve for the experiments (i.e. a concrete number when can compare across experiments). I see why you find this unclear, I will add some verbosity to the notebooks :)

File: relevance_maps_properties/metrics/metrics_text.ipynb (hunk 1 above)
Comment (WillemSpek) [reply relevance_maps_properties/metrics/metrics_text.ipynb:1]: Thanks, I experimented on both but only show one result in the notebook. Will factor in both LIME and RISE and compare their scores and visualizations in the notebook.

File: relevance_maps_properties/metrics/metrics_tutorial.ipynb (hunk 2 above)
Comment (WillemSpek) [reply relevance_maps_properties/metrics/metrics_tutorial.ipynb:1]: Thanks :)

**Comment by WillemSpek** [comment]
I have addressed your changes by making the notebooks far clearer :) Had to deal with some annoying merge conflicts because I created this branch from incremental deletion which received quite a few changes since its PR conversion, but all should be fine now. Feel free to review the changes.

**Review by cwmeijer — APPROVED** [review]
Looks great now! Thanks!

**Pull request merged**